### PR TITLE
Add atomic writes to local backend

### LINKED ISSUE #527: Request: Support atomic writes for local storage backend
It looks like there could be a risk of partially-written files with the local backend: https://github.com/chartmuseum/storage/blob/c3a87c79f56958041cb945ecf9706ed3da02e7f0/local.go#L102

That code appears to be writing directly to the real location of the file. So if the write is large and gets interrupted partway through (like chartmuseum is SIGKILL'd or if the file is copied/snapshotted by an external backup program), that could leave a partially-written file on disk.

Minio solves this problem by writing all files to a temporary directory, then atomically moving them into their real location after the write is fully complete: https://github.com/minio/minio/blob/d0862ddf866e6ac358155e3ca660f36610d8834e/cmd/fs-v1.go#L1105

This feature would allow users to backup their chartmuseum data the naive way (`cp`ing or `tar`ing all the files without having to stop/quiesce the application ahead of time).

Thoughts?

### PR DESCRIPTION
**Pull request by mac-chaffee**
Fixes #527

Also I noticed the tests were writing to `../../.test` which is outside the repo's directory when you run `make test`, so I moved that path back to `./.test`.

### CODE REVIEW

File: local.go [hunk 1]
```diff
@@ -35,7 +36,20 @@ func NewLocalFilesystemBackend(rootDirectory string) *LocalFilesystemBackend {
 	if err != nil {
 		panic(err)
 	}
-	b := &LocalFilesystemBackend{RootDirectory: absPath}
+	// Create a temporary folder for partially-completed writes (if not present)
+	tempPath := filepath.Join(absPath, ".tmp")
```
Comment (cbuto) [inline local.go:40]: this temp directory creation might be better suited in the `PutObject` function (like we do with the `RootDirectory`). This way the temp directory will always be recreated if it doesn't exist during a put, that way if that directory was deleted while chartmuseum is running, uploads will continue to work without restarting chartmuseum.

**Review by cbuto — CHANGES_REQUESTED** [review]

File: local.go (hunk 1 above)
Comment (mac-chaffee) [reply local.go:40]: That would require an extra `stat` per invocation of PutObject. Do we really want that performance overhead?

Comment (cbuto) [reply local.go:40]: that is definitely a consideration, but it seems like we should be treating the root dir and this new dir similarly. i.e. either both are created in `NewLocalFilesystemBackend` or both are created in `PutObject`. @jdolitsky @scbizu do you have any thoughts here?

Comment (scbizu) [reply local.go:40]: For @cbuto 's consideration , how about do anther `os.MkdirAll()` and ignore the dir already exist error in `PutObject` , I agree with cbuto's opinion and it is maybe reasonable even we do some overhead tradeoff here.

**Pull request closed without merging**